frontend/apps/reader/modules/readertypography.lua

```diff
@@ -0,0 +1,802 @@
+
+    -- Migrate old readerhyphenation settings (but keep them in case one
+    -- go back to a previous version)
+    if not G_reader_settings:readSetting("text_lang_default") and not G_reader_settings:readSetting("text_lang_fallback") then
+        local g_text_lang_set = false
+        local hyph_alg_default = G_reader_settings:readSetting("hyph_alg_default")
+        if hyph_alg_default then
+            local dict_info = HYPH_DICT_NAME_TO_LANG_TAG[hyph_alg_default]
+            if dict_info then
+                G_reader_settings:saveSetting("text_lang_default", dict_info[1])
+                g_text_lang_set = true
+                -- Tweak the other settings if the default hyph algo happens
+                -- to be one of these:
+                if hyph_alg_default == "@none" then
+                    G_reader_settings:saveSetting("hyphenation", false)
+                elseif hyph_alg_default == "@softhyphens" then
+                    G_reader_settings:saveSetting("hyph_soft_hyphens_only", true)
+                elseif hyph_alg_default == "@algorithm" then
+                    G_reader_settings:saveSetting("hyph_force_algorithmic", true)
+                end
+            end
+        end
+        local hyph_alg_fallback = G_reader_settings:readSetting("hyph_alg_fallback")
+        if not g_text_lang_set and hyph_alg_fallback then
+            local dict_info = HYPH_DICT_NAME_TO_LANG_TAG[hyph_alg_fallback]
+            if dict_info then
+                G_reader_settings:saveSetting("text_lang_fallback", dict_info[1])
+                g_text_lang_set = true
+                -- We can't really tweak other settings if the hyph algo fallback
+                -- happens to be @none, @softhyphens, @algortihm...
+            end
+        end
+        if not g_text_lang_set then
+            -- If nothing migrated, set the fallback to DEFAULT_LANG_TAG,
+            -- as we'll always have one of text_lang_default/_fallback set.
+            G_reader_settings:saveSetting("text_lang_fallback", DEFAULT_LANG_TAG)
+        end
+    end
+
+    local info_text = _([[
+Some languages have specific typographic rules: these include hyphenation, line breaking rules, and fonts language specific variations glyphs.
+KOReader will chose one according to the book metadata language tag, but you can select another one.
+You can also set a default language or a fallback one with a long-press.
+
+Features available per language are marked with:
+‐ : language specific hyphenation dictionary
+ : specific line breaking rules (mostly related to quotation marks)
+ : more line breaking rules (e.g. single letter prepositions not allowed at end of line)
+
+Note that when a language does not come with its own hyphenation dictionary, the English US one will be used.
+When the book language tag is not among our presets, no specific feature will be active, but it might be fine enough to trigger the fonts to use language variations known to them.]])
```

Comment:
book's
no specific features will be enabled
but it might be enough to get language specific glyph variants (when supported)